src/generate.ts

```diff
@@ -22,5 +28,101 @@ export function generate(specQ: SpecQuery, schema: Schema, opt: QueryConfig = DE
     }
   });
 
+  let ENCODING_QUERY_INDEX: Dict<EncodingQuery> = {};
+
+  if (opt.smallBandSizeForHighCardinality || opt.smallBandSizeForFacet) {
+    answerSet = answerSet.map(function(specQM) {
+
+      [Channel.ROW, Channel.Y, Channel.COLUMN, Channel.X].forEach((channel) => {
+        ENCODING_QUERY_INDEX[channel] = specQM.getEncodingQueryByChannel(channel);
+      });
+
+      const yEncQ = ENCODING_QUERY_INDEX[Channel.Y];
+      if (yEncQ !== undefined) {
+        if (ENCODING_QUERY_INDEX[Channel.ROW] ||
+            schema.cardinality(yEncQ) > 10) {
+
+          if (yEncQ.scale === undefined) {
+            specQM.setEncodingProperty(
```

Comment:
Like here you can just do

yEncQ.scale = yEncQ.scale || {};  
yEncQ.scale.bandSize = ...